src/components/Counter/Counter.jsx

```diff
@@ -1,11 +1,51 @@
-import { useEffect, useState } from 'react'
+import { useReducer, useState, useEffect } from 'react'
 
 const pinkRGB = `rgb(236, 72, 153)`
 
+const initailCount = 0
+
+function countReducer(count, action) {
+  switch (action.type) {
+    case 'increment': {
+      return count + 1
+    }
+    case 'decrement': {
+      return count - 1
+    }
+    case 'reset': {
+      return initailCount
+    }
+    default: {
+      throw Error(`Unknown action: ${action.type}`)
+    }
+  }
+}
+
 export default function Counter() {
-  const [count, setCount] = useState(0)
+  const [count, dispatch] = useReducer(countReducer, initailCount)
   const [currentColor, setCurrentColor] = useState(pinkRGB)
 
+  const handleIncrementCount = (increment) => {
```

Comment:
So for these cases, you don't actually need to pass in a parameter since all you are checking for is what type the action is and then running some code from your `switch`  for the associated type. If you were updating an object or state entry with a new object or had a payload that you needed to pass through as well then you could use an input parameter. 

In this use case, though, it's not necessary.